src/components/EventDashboard/CCPTabPanel.tsx

```diff
@@ -120,49 +120,117 @@ const EnhancedTableHead = (props: EnhancedTableProps) => {
 };
 
 const CCPTabPanel = ({ eventId }: { eventId: string }) => {
-  const [orderBy, setOrderBy] = React.useState<string>('name');
   const classes = useStyles();
-  const optionStyle = useOptions();
+  const { enqueueSnackbar } = useSnackbar();
+  const [orderBy, setOrderBy] = React.useState<string>('name');
   const [order, setOrder] = React.useState<Order>('asc');
-  const [row, selectRow] = React.useState<string | null>('');
-  const [anchorEl, setAnchorEl] = React.useState<
-    null | (EventTarget & HTMLButtonElement)
-  >(null);
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(
+    null
+  );
+  const [openConfirmDelete, setOpenConfirmDelete] = React.useState<boolean>(
+    false
+  );
+  const [selectedCCP, setSelectedCCP] = React.useState<CCP | null>(null);
+  const optionStyle = useOptions();
+
   const { data } = useQuery(GET_CCPS_BY_EVENT_ID, { variables: { eventId } });
 
   const rows = data ? data.collectionPointsByEvent : [];
   const open = Boolean(anchorEl);
   const history = useHistory();
 
+  const handleCloseMenu = () => {
+    setAnchorEl(null);
+    setSelectedCCP(null);
+  };
+
+  const handleCloseConfirmDelete = () => {
+    handleCloseMenu();
+    setOpenConfirmDelete(false);
+  };
+
+  //  Writing to cache when deleting ccp
+  const [deleteCCP] = useMutation(DELETE_CCP, {
+    update(cache, { data: { deleteCollectionPoint } }) {
+      if (!deleteCollectionPoint) {
+        return;
+      }
+
+      // Update GET_CCPS_BY_EVENT_ID
+      const { collectionPointsByEvent } = cache.readQuery<any>({
+        query: GET_CCPS_BY_EVENT_ID,
+        variables: { eventId },
+      });
+
+      let updatedCCPsList = collectionPointsByEvent.filter(
+        (ccp) => ccp.id !== (selectedCCP as CCP).id
+      );
+
+      cache.writeQuery({
+        query: GET_CCPS_BY_EVENT_ID,
+        variables: { eventId },
+        data: { collectionPointsByEvent: updatedCCPsList },
+      });
+
+      // Update GET_ALL_CCPS
+      const { collectionPoints } = cache.readQuery<any>({
+        query: GET_ALL_CCPS,
+      });
+
+      if (collectionPoints) {
+        updatedCCPsList = collectionPoints.filter(
+          (ccp) => ccp.id !== (selectedCCP as CCP).id
```

Comment:
Just curious, why did you do this: `(selectedCCP as CCP).id`?

Reply:
This is because on line 129: `const [selectedCCP, setSelectedCCP] = React.useState<CCP | null>(null);`, `selectedCCP` is defined as `CCP | null` and `null` has no property `id`. But if we know `selectedCCP` cannot be null when we reach this point in the code, so we're casting it as type `CCP`.

Reply:
Got it thanks!